apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -292,14 +337,34 @@ defmodule GameBackend.CurseOfMirra.Quests do
   defp filter_results_that_meet_quest_conditions(arena_match_results, conditions) do
     Enum.filter(arena_match_results, fn arena_match_result ->
       Enum.all?(conditions, fn condition ->
-        type = String.to_atom(condition["match_tracking_field"])
-        value = condition["value"]
-        arena_match_result_value = Map.get(arena_match_result, type)
+        case arena_match_result_field_to_atom(condition["match_tracking_field"]) do
+          # We'll result in a false value in case that the quest condition has a wrong field getter
+          nil ->
+            false
+
+          type ->
+            value = condition["value"]
+            arena_match_result_value = Map.get(arena_match_result, type)
 
-        comparator = parse_comparator(condition["comparison"])
+            comparator = parse_comparator(condition["comparison"])
 
-        comparator.(arena_match_result_value, value)
+            comparator.(arena_match_result_value, value)
+        end
       end)
     end)
   end
+
+  defp arena_match_result_field_to_atom("result"), do: :result
+  defp arena_match_result_field_to_atom("kills"), do: :kills
+  defp arena_match_result_field_to_atom("deaths"), do: :deaths
+  defp arena_match_result_field_to_atom("character"), do: :character
+  defp arena_match_result_field_to_atom("match_id"), do: :match_id
+  defp arena_match_result_field_to_atom("user_id"), do: :user_id
+  defp arena_match_result_field_to_atom("position"), do: :position
+  defp arena_match_result_field_to_atom("damage_done"), do: :damage_done
+  defp arena_match_result_field_to_atom("damage_taken"), do: :damage_taken
+  defp arena_match_result_field_to_atom("health_healed"), do: :health_healed
+  defp arena_match_result_field_to_atom("killed_by_bot"), do: :killed_by_bot
+  defp arena_match_result_field_to_atom("duration_ms"), do: :duration_ms
+  defp arena_match_result_field_to_atom(_), do: nil
```

Comment:
All these are defined here and in apps/arena/lib/arena/game/bounties.ex. Can you put them in some utils module or something?

Reply:
Oh one is for arena and the other one for game_backend, nvm.